Source/Events.Processing/Streams/IStreamProcessors.cs

```diff
@@ -12,21 +11,15 @@ namespace Dolittle.Runtime.Events.Processing.Streams
     /// </summary>
     public interface IStreamProcessors
     {
-        /// <summary>
-        /// Gets all registered stream processors.
-        /// </summary>
-        /// <returns>The<see cref="IEnumerable{StreamProcessor}" >stream processors</see>.</returns>
-        IEnumerable<StreamProcessor> Processors { get; }
-
         /// <summary>
         /// Registers a <see cref="StreamProcessor" />.
         /// </summary>
+        /// <param name="streamDefinition">The <see cref="StreamDefinition" /> of the stream the <see cref="StreamProcessor" /> will be registered on.</param>
         /// <param name="eventProcessor">The <see cref="IEventProcessor" />.</param>
         /// <param name="eventsFromStreamsFetcher">The <see cref="IFetchEventsFromStreams" />.</param>
-        /// <param name="sourceStreamId">The <see cref="StreamId" />.</param>
         /// <param name="cancellationToken">The <see cref="CancellationToken" />.</param>
-        /// <returns>The <see cref="StreamProcessorRegistrationResult"/>.</returns>
-        StreamProcessorRegistrationResult Register(IEventProcessor eventProcessor, IFetchEventsFromStreams eventsFromStreamsFetcher, StreamId sourceStreamId, CancellationToken cancellationToken);
+        /// <returns>The <see cref="StreamProcessorRegistration"/>.</returns>
+        StreamProcessorRegistration Register(StreamDefinition streamDefinition, IEventProcessor eventProcessor, IFetchEventsFromStreams eventsFromStreamsFetcher, CancellationToken cancellationToken);
```

Comment:
1) In the next PR this method returns a Task, so that won't be possible. And I don't really see the value of not returning a wrapped result
2) Hmm. I don't know if we want to couple the stream definition with a thing can can fetch events from that stream.

Reply:
We can follow up discussion later